client/src/components/Form/validation.ts

```diff
@@ -89,88 +94,87 @@ export const validationFunctions: ValidationFunctions<FormType> = {
+            (mfa) => mfa?.type === 'behandlingsdager',
+        ) as BehandlingsdagerPeriodeMFA[];
+        const reisetilskuddMFA = definedMFA.filter((mfa) => mfa?.type === 'reisetilskudd') as ReisetilskuddPeriodeMFA[];
+
+        // Perioder for avventende sykmelding
         if (
-            (schema.avventendeSykmelding && !schema.avventendePeriode) ||
-            (schema.avventendePeriode && schema.avventendePeriode.length === 1)
+            avventendeMFA.some(
+                (avventendeSykmelding) =>
+                    !avventendeSykmelding.avventendePeriode || avventendeSykmelding.avventendePeriode.length === 1,
+            )
         ) {
-            return 'Periode må være definert når avventende sykmelding er krysset av';
+            // TODO: This check will never occur as RangePicker will set end date to start date if no end date is selected by the user. Currently not an issue as this allows the user to set a period of 1 day
+            return 'Periode må være definert når avventende sykmelding er valgt';
         }
-    },
-    avventendeInnspillTilArbeidsgiver: (schema) => {
-        if (schema.avventendeSykmelding && !schema.avventendeInnspillTilArbeidsgiver) {
+
+        if (avventendeMFA.some((avventendeSykmelding) => !avventendeSykmelding.avventendeInnspillTilArbeidsgiver)) {
             return 'Innspill til arbeidsgiver om tilrettelegging må være utfylt når avventende sykmelding er krysset av';
         }
-    },
-    // Perioder for gradert sykmelding
-    gradertSykmelding: () => undefined,
-    gradertPeriode: (schema) => {
+
+        // Perioder for gradert sykmelding
         if (
-            (schema.gradertSykmelding && !schema.gradertPeriode) ||
-            (schema.gradertPeriode && schema.gradertPeriode.length === 1)
+            gradertMFA.some(
+                (gradertSykmelding) =>
+                    !gradertSykmelding.gradertPeriode || gradertSykmelding.gradertPeriode.length === 1,
+            )
         ) {
-            return 'Periode må være definert når gradert sykmelding er krysset av';
+            return 'Periode må være definert når gradert sykmelding er valgt';
         }
-    },
-    gradertGrad: () => undefined,
-    gradertReisetilskudd: () => undefined,
-    // Perioder for full sykmelding
-    aktivitetIkkeMuligSykmelding: () => undefined,
-    aktivitetIkkeMuligPeriode: (schema) => {
+
+        // Perioder for full sykmelding
         if (
-            (schema.aktivitetIkkeMuligSykmelding && !schema.aktivitetIkkeMuligPeriode) ||
-            (schema.aktivitetIkkeMuligPeriode && schema.aktivitetIkkeMuligPeriode.length === 1)
+            aktivitetIkkeMuligMFA.some(
+                (aktivitetIkkeMulig) =>
+                    !aktivitetIkkeMulig.aktivitetIkkeMuligPeriode ||
+                    aktivitetIkkeMulig.aktivitetIkkeMuligPeriode.length === 1,
+            )
         ) {
             return 'Periode må være definert når aktivitet ikke er mulig';
         }
```

Comment:
```suggestion
        }
        
                if (
            aktivitetIkkeMuligMFA.some(
                (aktivitetIkkeMulig) =>
                    aktivitetIkkeMulig.aktivitetIkkeMuligMedisinskArsak &&
                    aktivitetIkkeMulig.aktivitetIkkeMuligMedisinskArsakType?.length === 0,
            )
        ) {
            return 'Minst én medisinsk årsak må være valgt når det er medisinske årsaker som hindrer aktivitet';
        }

        if (
            aktivitetIkkeMuligMFA.some(
                (aktivitetIkkeMulig) =>
                    aktivitetIkkeMulig.aktivitetIkkeMuligMedisinskArsak &&
                    aktivitetIkkeMulig.aktivitetIkkeMuligMedisinskArsakType?.length === 0,
            )
        ) {
            return 'Minst én arbeidsrelatert årsak må være valgt når det er arbeidsrelaterte årsaker som hindrer aktivitet';
        }
```

Reply:
Fant en bug knyttet til dette som jeg ikke forstår.
Prøv å submit med den mockdataen som ligger inne uten å endre noe. Da vil du få feilen "Minsk én medisinsk årask må være valgt når det er medi....."

Hvis du så:
1. Sletter alle perioder
2. Legger til ny periode
3. velger 100% sykmelding
4. angir periode
5. checker av "Det er medisinske..." uten å velge årsak.
6. submitter
-> Da vil ikke samme feilen trigge selv om den burde gjøre det

Reply:
Løst, aktivitetIkkeMuligMedisinskArsakType gikk fra å ha verdien [] til undefined, og vi sjekket ikke for undefined i valideringen